resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/repository/AbstractReactiveRepositoryRedisAdapter.java

```diff
@@ -270,38 +294,41 @@ public ResourceRepositoryAdapter(final ReactiveRedisTemplate<String, ResourceEnt
         @Override
         @NonNull
         public <S extends ResourceEntity> Mono<S> save(final @NonNull S entity) {
-            return this.setOps.add(this.table + KEY_SEP + entity.getParentId(), entity.getId())
-                    .then(super.saveDefault(entity));
+            final String id = this.table + SEPARATOR + entity.getParentId();
+            return this.setOps.add(id, entity.getId())
+                    .then(this.redisTemplate.expire(id, ttl))
+                    .then(this.saveDefault(entity));
         }
 
         @Override
         public Flux<ResourceEntity> findAllByParentId(final String parentId) {
-            return this.setOps.members(this.table + KEY_SEP + parentId)
+            return this.setOps.members(this.table + SEPARATOR + parentId)
                     .flatMap(this::findById);
         }
 
         @Override
         @NonNull
         public Mono<Void> deleteById(final @NonNull String key) {
             return this.findById(key)
-                    .map(entity -> this.setOps.remove(this.table + KEY_SEP + entity.getParentId()))
-                    .then(super.deleteByIdDefault(key));
+                    .map(entity -> this.setOps.remove(this.table + SEPARATOR + entity.getParentId()))
+                    .then(this.deleteByIdDefault(key));
         }
     }
 
     /**
      * A class to allow {@link SerialisedFormatEntity}s to be stored in a reactive repository
      */
     public static class SerialisedFormatRepositoryAdapter extends AbstractReactiveRepositoryRedisAdapter<SerialisedFormatEntity, String> implements SerialisedFormatRepository {
+        private static final String SEPARATOR = KEY_SEP + FORMAT_KEYSPACE + KEY_SEP;
```

Comment:
I'd rename this to `SERIALISED_FORMAT_SEP` or similar

Reply:
Renamed